projects/components/src/table/columns/table-edit-columns-modal.component.ts

```diff
@@ -47,7 +47,15 @@ export class TableEditColumnsModalComponent {
     private readonly modalRef: ModalRef<TableColumnConfigExtended[]>,
     @Inject(MODAL_DATA) public readonly modalData: TableColumnConfigExtended[]
   ) {
-    this.editColumns = this.modalData.filter(column => (column.attribute?.type as string) !== '$$state');
+    const checked = this.modalData
+      .filter(column => (column.attribute?.type as string) !== '$$state')
+      .filter(column => column.visible);
+
+    const unchecked = this.modalData
+      .filter(column => (column.attribute?.type as string) !== '$$state')
+      .filter(column => !column.visible);
+
+    this.editColumns = [...checked, ...unchecked];
```

Comment:
Is the purpose of this split to just sort by checked status? If yes, would be clearer to just use sort.

Reply:
Really? My first iteration was using sort but that seemed less clear since the sort was a little ugly.

Reply:
hmm I was imagining a sort taking advantage of coercion like:
```typescript
sort((first, second) => Number(first.visible) - Number(second.visible))
```
or more explicitly
```typescript
sort((first, second) => first.visible === second.visible ? 0 : first.visible ? -1 : 1)
```
Not a big deal, whichever of the three approaches you think is clearer.

Reply:
I think I had something like your second example originally. I'm fine going back to that. Will update.